Session 2/Functioning/IteratorsAndSizing.cpp

```diff
@@ -0,0 +1,50 @@
+#include<iostream>
+#include<vector>
+using namespace std;
+void disp(vector<int> &x){
+    cout<<"\nSize - "<<x.size();
+    cout<<"\nCapacity - "<<x.capacity();
+}
+int main(){
+    vector<int> v;
+    disp(v);
+
+    //1# Illustration of how vectors allows the size flexibility.
+    v.reserve(4);//Requests that the vector capacity be at least enough to contain n elements.
+    disp(v);
+    for(int i=0;i<3;i++)
+        v.push_back(i*2);
+    disp(v);
+    v.push_back(6);
+    disp(v);
+    v.push_back(8);
+    disp(v);
+    //how to shrink the capacity of vector equal to the size of vector
+    //v.shrink_to_fit();
+    //disp(v);
+    v.resize(7); // size increases to 7 - {0,2,4,6,8,0,0}
+    v.resize(10,5);//all the new elements are initialized with value 5 - {0,2,4,6,8,0,0,5,5,5}
+
+
+    //2# Element Access using reference functions - these functions returns the reference of element
+    cout<<endl;
+    cout<<v[4]<<"\n";
+    cout<<v.front()<<"\n";// returns the first element
+    cout<<v.back()<<"\n";//returns the last element; end() != back()
+    cout<<v.at(4)<<"\n";//returns the reference of argument index at(int index);
+
```

Comment:
I think these functions return the "value" at front, back and at.
Whereas begin() and end() return a reference iterator pointing at the beginning and end respectively